backends/p4test/run-p4-sample.py

```diff
@@ -336,8 +346,10 @@ def main(argv):
                 argv = argv[1:]
         elif argv[0][1] == 'D' or argv[0][1] == 'I' or argv[0][1] == 'T':
             options.compilerOptions.append(argv[0])
-        elif argv[0] == "-gdb":
+        elif argv[0][0:4] == "-gdb":
```

Comment:
what happens if this is not long enough?

Reply:
If the argument is shorter, the slice will get the entire arg (so the test will be false, but that is correct)